helix-term/src/commands.rs

```diff
@@ -1839,73 +1882,78 @@ mod cmd {
         pub aliases: &'static [&'static str],
         pub doc: &'static str,
         // params, flags, helper, completer
-        pub fun: fn(&mut compositor::Context, &[&str], PromptEvent) -> anyhow::Result<()>,
+        pub fun: fn(&mut Editor, &mut Jobs, &[&str], PromptEvent) -> anyhow::Result<()>,
         pub completer: Option<Completer>,
     }
 
     fn quit(
-        cx: &mut compositor::Context,
+        editor: &mut Editor,
+        _jobs: &mut Jobs,
```

Comment:
Why was the context removed ? Borrow checker problems ?

Reply:
The context provided by the `execute` method in the `MappableCommand` is the `Context<'a>`  struct defined within the `command.rs` module. This method was previously used only for static commands but is now used to call both types of command. The context used previously for the `TypableCommands` is actually the `compositor::Context` which is a different type all together. They both include `Editor` and `Jobs` which are the two things all of the `TypableCommands` require so I simply just replaced the compositor Context with the actual fields needed.

Reply:
I'd rather see that you keep the signature and just construct a compositor::Context